### PR TITLE
Add strict mode to DefinitionStorage

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | https://github.com/yiisoft/di/pull/278

### CODE REVIEW

**Review by rustamwin — APPROVED** [review]

File: tests/Unit/DefinitionStorageTest.php [hunk 1]
```diff
@@ -2,16 +2,20 @@
 
 declare(strict_types=1);
 
-namespace Yiisoft\Dfinitions\Helpers\Tests\Unit;
+namespace Yiisoft\Dfinitions\Tests\Unit;
 
+use NonExisitng;
+use NonExisting;
```
Comment (vjik) [inline tests/Unit/DefinitionStorageTest.php:8]:
```suggestion
```

File: tests/Unit/DefinitionStorageTest.php [hunk 2]
```diff
@@ -105,10 +109,26 @@ public function testServiceWithNonExistingUnionTypes(): void
...
+        $this->assertTrue($storage->has(EngineMarkOne::class));
+        $this->assertSame(EngineMarkOne::class, $storage->get(EngineMarkOne::class));
+    }
+
+    public function testStrictModeEnabled(): void
+    {
+        $storage = new DefinitionStorage([], true);
+        $this->assertFalse($storage->has(EngineMarkOne::class));
+
+        $this->expectException(RuntimeException::class);
```
Comment (vjik) [inline tests/Unit/DefinitionStorageTest.php:131]:
```suggestion
        $this->expectException(RuntimeException::class);
        $this->expectExceptionMessage('Service ' . EngineMarkOne::class . ' doesn\'t exist in DefinitionStorage.');
```

File: src/DefinitionStorage.php [hunk 3]
```diff
@@ -89,6 +95,10 @@ private function isResolvable(string $id, array $building): bool
             return true;
         }
 
+        if ($this->useStrictMode) {
+            return false;
```
Comment (vjik) [inline src/DefinitionStorage.php:99]: Should we add ID into build stack here?

**Review by vjik — CHANGES_REQUESTED** [review]
If use `DefinitionStorage::get()` wthout run before `DefinitionStorage::has()`, then storage always work as strict. Should it be like this?

**Pull request merged**